app/src/main/java/com/example/drawingactivity/DrawableAreaView.kt

```diff
@@ -0,0 +1,81 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+
+@SuppressLint("ClickableViewAccessibility")
+class DrawableAreaView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    var onBitmapDrawn: ((Bitmap) -> Unit)? = null
+    var bitmap: Bitmap? = null
+    private var shape: Rect? = null
+
+    var pathColor = Color.BLACK
+    private var pathPaint = Paint().apply {
+        color = pathColor
+        style = Paint.Style.STROKE
+        strokeWidth = 20f
+        strokeJoin = Paint.Join.ROUND
+        strokeCap = Paint.Cap.ROUND
+    }
+
+    private var bitmapPaint = Paint()
+    lateinit var path: Path
+
+    init {
+        setOnTouchListener { _, mEvent ->
+            when (mEvent.action) {
+                MotionEvent.ACTION_DOWN -> {
+                    pathPaint.color = pathColor
+                    path = Path()
+                    path.moveTo(mEvent.x, mEvent.y)
+                    invalidate()
+                }
+                MotionEvent.ACTION_MOVE -> {
+                    path.lineTo(mEvent.x, mEvent.y)
+                    invalidate()
+                }
+                MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> {
```

Comment:
Yeah we should be drawing the path to the bitmap after making it, so that the bitmap has a representation of each path. EACH path. Is this what the lambda is for?.... Investigating

Reply:
lambda has been removed as it was for saving and loading the bitmap from the viewModel on orientation change. I couldn't quite understand how to fix it but when the canvas initialized it would overwrite the bitmap the viewModel gave it with the null it would set the variable to.